packages/config-resolver/src/regionInfo/getHostnameFromVariants.ts

```diff
@@ -0,0 +1,14 @@
+import { EndpointVariant } from "./EndpointVariant";
+
+export interface GetHostnameFromVariantsOptions {
+  isFipsEndpoint: boolean;
+  isDualstackEndpoint: boolean;
+}
+
+export const getHostnameFromVariants = (
+  variants: EndpointVariant[] = [],
+  { isFipsEndpoint, isDualstackEndpoint }: GetHostnameFromVariantsOptions
+) =>
```

Comment:
Explicitly define, as by default TS thinks it's string
```suggestion
): string | undefined => 
```